src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1496,7 +1496,13 @@ def serialize_key_and_certificates_to_pkcs12(
     def poly1305_supported(self) -> bool:
         if self._fips_enabled:
             return False
-        return self._lib.Cryptography_HAS_POLY1305 == 1
+        elif (
+            self._lib.CRYPTOGRAPHY_IS_BORINGSSL
+            or self._lib.CRYPTOGRAPHY_IS_LIBRESSL
+        ):
+            return True
+        else:
+            return self._lib.Cryptography_HAS_POLY1305 == 1
```

Comment:
I wonder if it makes sense to compute `Cryptography_HAS_POLY1305` based on `CRYPTOGRAPHY_IS_BORINGSSL` and `CRYPTOGRAPHY_IS_LIBRESSL`, rather than special-casing them here? Not sure.

Reply:
Maybe. I didn't do it because `Cryptography_HAS_POLY1305` is computed in [nid.py](https://github.com/pyca/cryptography/blob/34c2f024141e40232bf1de238bd6a4b8480cefa7/src/_cffi_src/openssl/nid.py#L40-L44) based on `NID_poly1305`, which is OpenSSL-specific (Boring and Libre do not have a NID for Poly1305).
So it was mostly me not wanting to put not-NID-related code inside `nid.py`. But if that's OK then we can do it there.

Reply:
That makes sense to me. We'll see what @alex thinks 🙂

Reply:
I think this is probably fine as-is -- if we changed `Cryptography_HAS_POLY1305` to be true on libre/boring, we'd need a new name for its existing purpose.